Task_2_1_1/src/main/java/ru/nsu/voronova/ThreadPrimeSearch.java

```diff
@@ -0,0 +1,42 @@
+package ru.nsu.voronova;
+
+import java.util.*;
+import java.util.concurrent.*;
+
+public class ThreadPrimeSearch extends PrimeSearch {
+    private final int THREADS_NUMBER = 10;
+
+    @Override
+    public boolean hasPrime(int[] array) throws ExecutionException, InterruptedException, NullPointerException {
+        if (array == null) {
+            throw new NullPointerException();
+        }
+        List<Integer> numberList = Arrays.stream(array).boxed().toList();
+        Deque<Integer> deque = new ArrayDeque<>(numberList);
```

Comment:
They are not thread-safe; in the absence of external synchronization, they do not support concurrent access by multiple threads.